src/wrap-store/wrapStore.js

```diff
@@ -71,25 +74,33 @@ export default (store, {
       return;
     }
 
-    /**
-     * Send store's current state through port
-     * @return undefined
-     */
-    const sendState = () => {
-      port.postMessage({
-        type: STATE_TYPE,
-        payload: store.getState()
-      });
+    let prevState = store.getState();
+
+    const patchState = () => {
```

Comment:
This is where I was getting things wrong in my head. I like how you're doing the shallow diffing on the background page so the content script doesn't have to check equality at all (since all things serialized down will be new). Good shit. I need to make sure to grab a cup of ☕️ before talking in GH issues. :-P